### PR TITLE
[in_app_purchase]  Migrate to NNBD

### PR DESCRIPTION
**Pull request by cyanglaz**
Migrate the plugin to null safety.
Also cleaned up some redundant deps

Fixes https://github.com/flutter/flutter/issues/75237
Also fixes https://github.com/flutter/flutter/issues/69225

While migrating, I also found that there's a serious bad naming and documentation of a parameter (sandboxTesting) so I decided to deprecated as well. Please let me know if you want it reverted and in a different PR. 
This bad naming caused a lot confusion and people has been commenting in issues about it, for example: 
https://github.com/flutter/flutter/issues/66253#issuecomment-716115879

## Pre-launch Checklist

- [x] The title of the PR starts with the name of the plugin surrounded by square brackets, e.g. `[shared_preferences]`
- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide] and the [C++, Objective-C, Java style guides].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test exempt.
- [x] I updated pubspec.yaml with an appropriate new version according to the [pub versioning philosophy].
- [x] I updated CHANGELOG.md to add a description of the change.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I signed the [CLA].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[C++, Objective-C, Java style guides]: https://github.com/flutter/engine/blob/master/CONTRIBUTING.md#style
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning

### CODE REVIEW

File: packages/in_app_purchase/lib/src/in_app_purchase/purchase_details.dart [hunk 1]
```diff
@@ -103,10 +103,16 @@ class PurchaseParam {
...
 
-  /// The 'sandboxTesting' is only available on iOS, set it to `true` for testing in AppStore's sandbox environment. The default value is `false`.
+  /// @deprecated Use [simulatesAskToBuyInSandbox] instead.
+  ///
+  /// This parameter has no effect and will be removed shortly.
+  @deprecated
   final bool sandboxTesting;
+
+  /// Only available on iOS, set it to `true` to produce an "ask to buy" flow for this payment in the sandbox.
+  final bool simulatesAskToBuyInSandbox;
```
Comment (cyanglaz) [inline packages/in_app_purchase/lib/src/in_app_purchase/purchase_details.dart:115]: CC @LHLL 

This is the rename I talked about.

File: packages/in_app_purchase/build.yaml [hunk 2]
```diff
@@ -5,4 +5,3 @@ targets:
         options:
           any_map: true
           create_to_json: true
-          nullable: false
```
Comment (cyanglaz) [inline packages/in_app_purchase/build.yaml:8]: This is deprecated, so removing

File: packages/in_app_purchase/example/lib/main.dart [hunk 3]
```diff
@@ -2,6 +2,7 @@
 // Use of this source code is governed by a BSD-style license that can be
 // found in the LICENSE file.
 
+// @dart = 2.10
```
Comment (stuartmorgan) [inline packages/in_app_purchase/example/lib/main.dart:5]: We should be migrating the example at the same time, per recent discussion.

File: packages/in_app_purchase/integration_test/in_app_purchase_test.dart [hunk 4]
```diff
@@ -1,6 +1,7 @@
 // Copyright 2019, the Chromium project authors.  Please see the AUTHORS file
 // for details. All rights reserved. Use of this source code is governed by a
 // BSD-style license that can be found in the LICENSE file.
+// @dart = 2.10
```
Comment (stuartmorgan) [inline packages/in_app_purchase/integration_test/in_app_purchase_test.dart:4]: Any reason you're using 2.10, rather than the 2.9 that's suggested in the unsound NNBD docs?

File: packages/in_app_purchase/CHANGELOG.md [hunk 5]
```diff
@@ -1,3 +1,10 @@
+## 0.4.0-nullsafety.0
+
+* Migrate to nullsafety.
+* Deprecate `sandboxTesting`, introduce `simulatesAskToBuyInSandbox`.
+* **Breaking Change:**
+  * Removed `callbackChannel` in `channels.dart`, see https://github.com/flutter/flutter/issues/69225.
```
Comment (stuartmorgan) [inline packages/in_app_purchase/CHANGELOG.md:6]: What does this break? Is it just channel-based mocks in tests?

File: packages/in_app_purchase/ios/Classes/InAppPurchasePlugin.m [hunk 6]
```diff
@@ -290,7 +290,7 @@ - (void)refreshReceipt:(FlutterMethodCall *)call result:(FlutterResult)result {
   }];
 }
 
-#pragma mark - delegates
+#pragma mark - delegatestransactionIdentifier:
```
Comment (stuartmorgan) [inline packages/in_app_purchase/ios/Classes/InAppPurchasePlugin.m:293]: Is this a mistake? This doesn't read like a section title any more.

File: packages/in_app_purchase/lib/src/billing_client_wrappers/billing_client_wrapper.dart [hunk 7]
```diff
@@ -74,8 +73,13 @@ class BillingClient {
...
   /// [`BillingClient#isReady()`](https://developer.android.com/reference/com/android/billingclient/api/BillingClient.html#isReady())
   /// to get the ready status of the BillingClient instance.
-  Future<bool> isReady() async =>
-      await channel.invokeMethod<bool>('BillingClient#isReady()');
+  Future<bool> isReady() async {
+    bool? ready = await channel.invokeMethod<bool>('BillingClient#isReady()');
+    if (ready == null) {
+      return false;
+    }
+    return ready;
```
Comment (stuartmorgan) [inline packages/in_app_purchase/lib/src/billing_client_wrappers/billing_client_wrapper.dart:81]: `return ready ?? false`?

File: packages/in_app_purchase/lib/src/billing_client_wrappers/billing_client_wrapper.dart [hunk 8]
```diff
@@ -100,20 +104,20 @@ class BillingClient {
...
     disconnectCallbacks.add(onBillingServiceDisconnected);
-    return BillingResultWrapper.fromJson(await channel
+    return BillingResultWrapper.fromJson((await channel
         .invokeMapMethod<String, dynamic>(
             "BillingClient#startConnection(BillingClientStateListener)",
             <String, dynamic>{
           'handle': disconnectCallbacks.length - 1,
           'enablePendingPurchases': _enablePendingPurchases
-        }));
+        }))!);
```
Comment (stuartmorgan) [inline packages/in_app_purchase/lib/src/billing_client_wrappers/billing_client_wrapper.dart:120]: Are we positive there are no codepaths in any of the native code that return null for all these calls you're asserting?

File: packages/in_app_purchase/lib/src/billing_client_wrappers/billing_client_wrapper.dart [hunk 9]
```diff
@@ -283,15 +288,15 @@ class BillingClient {
...
       case kOnPurchasesUpdated:
         // The purchases updated listener is a singleton.
-        assert(_callbacks[kOnPurchasesUpdated].length == 1);
+        assert(_callbacks[kOnPurchasesUpdated]!.length == 1);
         final PurchasesUpdatedListener listener =
-            _callbacks[kOnPurchasesUpdated].first;
+            _callbacks[kOnPurchasesUpdated]!.first as PurchasesUpdatedListener;
         listener(PurchasesResultWrapper.fromJson(
-            call.arguments.cast<String, dynamic>()));
+            Map<String, dynamic>.from(call.arguments)));
```
Comment (stuartmorgan) [inline packages/in_app_purchase/lib/src/billing_client_wrappers/billing_client_wrapper.dart:295]: Why the change here?

File: packages/in_app_purchase/lib/src/billing_client_wrappers/purchase_wrapper.dart [hunk 10]
```diff
@@ -74,46 +75,55 @@ class PurchaseWrapper {
...
+  @JsonKey(defaultValue: '')
   final String sku;
 
   /// True for subscriptions that renew automatically. Does not apply to
   /// [SkuType.inapp] products.
   ///
   /// For [SkuType.subs] this means that the subscription is canceled when it is
   /// false.
-  final bool isAutoRenewing;
+  final bool? isAutoRenewing;
```
Comment (stuartmorgan) [inline packages/in_app_purchase/lib/src/billing_client_wrappers/purchase_wrapper.dart:107]: What would a null value mean here? If this is intentional, it should be documented.

File: packages/in_app_purchase/lib/src/in_app_purchase/in_app_purchase_connection.dart [hunk 11]
```diff
@@ -298,9 +300,9 @@ class IAPError {
   /// The error code.
   final String code;
 
-  /// A human-readable error message, possibly null.
+  /// A human-readable error message.
```
Comment (stuartmorgan) [inline packages/in_app_purchase/lib/src/in_app_purchase/in_app_purchase_connection.dart:303]: Why the change here?

File: packages/in_app_purchase/lib/src/in_app_purchase/purchase_details.dart [hunk 12]
```diff
@@ -103,10 +103,16 @@ class PurchaseParam {
   /// Do not pass in a clear text, your developer ID, the user’s Apple ID, or the
   /// user's Google ID for this field.
   /// For example, you can use a one-way hash of the user’s account name on your server.
-  final String applicationUserName;
+  final String? applicationUserName;
 
-  /// The 'sandboxTesting' is only available on iOS, set it to `true` for testing in AppStore's sandbox environment. The default value is `false`.
+  /// @deprecated Use [simulatesAskToBuyInSandbox] instead.
+  ///
+  /// This parameter has no effect and will be removed shortly.
```
Comment (stuartmorgan) [inline packages/in_app_purchase/lib/src/in_app_purchase/purchase_details.dart:110]: Shouldn't it have exactly the same effect as simulatesAskToBuyInSandbox? Otherwise you're not really deprecating it, you're breaking it, which seems strictly worse than just removing it entirely.

File: packages/in_app_purchase/lib/src/in_app_purchase/purchase_details.dart [hunk 13]
```diff
@@ -126,15 +134,16 @@ class PurchaseDetails {
...
+  /// On iOS, [InAppPurchaseConnection.refreshPurchaseVerificationData] can be used to get a new
   /// [PurchaseVerificationData] object for further validation.
   final PurchaseVerificationData verificationData;
 
   /// The timestamp of the transaction.
   ///
   /// Milliseconds since epoch.
-  final String transactionDate;
+  ///
+  /// The value is `null` is [status] is not [PurchaseStatus.purchased].
```
Comment (stuartmorgan) [inline packages/in_app_purchase/lib/src/in_app_purchase/purchase_details.dart:145]: Typo: s/is [status]/if [status]/

File: packages/in_app_purchase/lib/src/store_kit_wrappers/sk_payment_queue_wrapper.dart [hunk 14]
```diff
@@ -124,19 +123,20 @@ class SKPaymentQueueWrapper {
...
     await channel.invokeMethod<void>(
         '-[InAppPurchasePlugin restoreTransactions:result:]',
         applicationUserName);
   }
 
   // Triage a method channel call from the platform and triggers the correct observer method.
-  Future<dynamic> _handleObserverCallbacks(MethodCall call) {
+  Future<void> _handleObserverCallbacks(MethodCall call) async {
     assert(_observer != null,
         '[in_app_purchase]: (Fatal)The observer has not been set but we received a purchase transaction notification. Please ensure the observer has been set using `setTransactionObserver`. Make sure the observer is added right at the App Launch.');
```
Comment (stuartmorgan) [inline packages/in_app_purchase/lib/src/store_kit_wrappers/sk_payment_queue_wrapper.dart:141]: Consider setting a local `observer` variable so you're using `!` once where it's obvious why it's okay, rather than throughout  a somewhat long method.

File: packages/in_app_purchase/lib/src/store_kit_wrappers/sk_product_wrapper.dart [hunk 15]
```diff
@@ -20,28 +19,29 @@ part 'sk_product_wrapper.g.dart';
 class SkProductResponseWrapper {
   /// Creates an [SkProductResponseWrapper] with the given product details.
   SkProductResponseWrapper(
-      {@required this.products, @required this.invalidProductIdentifiers});
+      {required this.products, required this.invalidProductIdentifiers});
 
   /// Constructing an instance from a map from the Objective-C layer.
   ///
   /// This method should only be used with `map` values returned by [SKRequestMaker.startProductRequest].
   /// The `map` parameter must not be null.
```
Comment (stuartmorgan) [inline packages/in_app_purchase/lib/src/store_kit_wrappers/sk_product_wrapper.dart:27]: Remove this comment; it's now redundant.

File: packages/in_app_purchase/pubspec.yaml [hunk 16]
```diff
@@ -1,30 +1,26 @@
...
+version: 0.4.0-nullsafety.0
 
 dependencies:
-  async: ^2.0.8
-  collection: ^1.14.11
   flutter:
     sdk: flutter
-  json_annotation: ^3.0.0
-  meta: ^1.1.6
+  json_annotation: ^4.0.0-nullsafety.0
```
Comment (stuartmorgan) [inline packages/in_app_purchase/pubspec.yaml:9]: Just 4.0.0 now. In fact, it looks like all of the `-nullsafe` versions here now have stable previews you can use.

File: packages/in_app_purchase/pubspec.yaml [hunk 17]
```diff
@@ -1,30 +1,26 @@
 name: in_app_purchase
 description: A Flutter plugin for in-app purchases. Exposes APIs for making in-app purchases through the App Store and Google Play.
 homepage: https://github.com/flutter/plugins/tree/master/packages/in_app_purchase
-version: 0.3.5+2
+version: 0.4.0-nullsafety.0
```
Comment (stuartmorgan) [inline packages/in_app_purchase/pubspec.yaml:4]: Looks like you can just do 0.4.0 and avoid a second pass to publish this as stable later.

File: packages/in_app_purchase/test/store_kit_wrappers/sk_product_test.dart [hunk 18]
```diff
@@ -17,19 +17,10 @@ void main() {
...
       expect(wrapper, equals(dummySubscription));
     });
 
-    test(
-        'SKProductSubscriptionPeriodWrapper should have properties to be null if map is empty',
-        () {
-      final SKProductSubscriptionPeriodWrapper wrapper =
-          SKProductSubscriptionPeriodWrapper.fromJson(<String, dynamic>{});
-      expect(wrapper.numberOfUnits, null);
-      expect(wrapper.unit, null);
```
Comment (stuartmorgan) [inline packages/in_app_purchase/test/store_kit_wrappers/sk_product_test.dart:30]: For these removed tests, shouldn't we be able to assert that everything has expected default values if the map is empty, rather than removing the tests?

**Review by stuartmorgan — CHANGES_REQUESTED** [review]

File: packages/in_app_purchase/lib/src/in_app_purchase/purchase_details.dart (hunk 1 above)
Comment (LHLL) [reply packages/in_app_purchase/lib/src/in_app_purchase/purchase_details.dart:115]: I think Apple's documentation is very opaque, do we want to make it specific here? Maybe something like "Setting it to `true` will cause a transaction to be in the state `SKPaymentTransactionStateDeferred` and produce an "ask to buy" prompt that interrupts the the payment flow."

File: packages/in_app_purchase/lib/src/in_app_purchase/in_app_purchase_connection.dart (hunk 11 above)
Comment (cyanglaz) [reply packages/in_app_purchase/lib/src/in_app_purchase/in_app_purchase_connection.dart:303]: I think empty error message and null error message means the same thing, so I thought we could make this non null so we don't have 2 values mean the same thing.

File: packages/in_app_purchase/lib/src/in_app_purchase/purchase_details.dart (hunk 12 above)
Comment (cyanglaz) [reply packages/in_app_purchase/lib/src/in_app_purchase/purchase_details.dart]: Redocument this and make this the same effect as simulatesAskToBuyInSandbox.

File: packages/in_app_purchase/lib/src/in_app_purchase/purchase_details.dart (hunk 1 above)
Comment (cyanglaz) [reply packages/in_app_purchase/lib/src/in_app_purchase/purchase_details.dart:119]: Done. I added the detailed description in `sk_payment_queue_wrapper.dart` and add a link here.

File: packages/in_app_purchase/lib/src/in_app_purchase/purchase_details.dart (hunk 13 above)
Comment (cyanglaz) [reply packages/in_app_purchase/lib/src/in_app_purchase/purchase_details.dart]: done

File: packages/in_app_purchase/lib/src/store_kit_wrappers/sk_payment_queue_wrapper.dart (hunk 14 above)
Comment (cyanglaz) [reply packages/in_app_purchase/lib/src/store_kit_wrappers/sk_payment_queue_wrapper.dart:141]: done!

File: packages/in_app_purchase/lib/src/store_kit_wrappers/sk_product_wrapper.dart (hunk 15 above)
Comment (cyanglaz) [reply packages/in_app_purchase/lib/src/store_kit_wrappers/sk_product_wrapper.dart]: done, and removed all the similar documentations.

File: packages/in_app_purchase/pubspec.yaml (hunk 16 above)
Comment (cyanglaz) [reply packages/in_app_purchase/pubspec.yaml]: done

File: packages/in_app_purchase/pubspec.yaml (hunk 17 above)
Comment (cyanglaz) [reply packages/in_app_purchase/pubspec.yaml]: done

**Comment by cyanglaz** [comment]
@stuartmorgan 

I tried to migrate the example app but got some error related to shared_prefernece nnbd
```
E/flutter ( 5242): [ERROR:flutter/lib/ui/ui_dart_state.cc(186)] Unhandled Exception: type 'List<dynamic>' is not a subtype of type 'List<Object>?' in type cast
E/flutter ( 5242): #0      SharedPreferences.getStringList (package:shared_preferences/shared_preferences.dart:110:48)
E/flutter ( 5242): #1      ConsumableStore.load (package:in_app_purchase_example/consumable_store.dart:34:52)
E/flutter ( 5242): <asynchronous suspension>
E/flutter ( 5242): #2      _MyAppState.initStoreInfo (package:in_app_purchase_example/main.dart:116:32)
E/flutter ( 5242): <asynchronous suspension>
E/flutter ( 5242): 
```

Looks like an issue related to the shared_preference. Should we land this without migrating the example app and fix it in the shared_preference first?

**Comment by cyanglaz** [comment]
Review comments are mostly fixed. I'm waiting on the new shared_preference fix to be published to do a final round of testing, then I'll ping again for review.

File: packages/in_app_purchase/integration_test/in_app_purchase_test.dart (hunk 4 above)
Comment (blasten) [reply packages/in_app_purchase/integration_test/in_app_purchase_test.dart:4]: We used 2.9 in the rest of the plugins, so I'm not sure if I suggested 2.10

Comment (cyanglaz) [reply packages/in_app_purchase/integration_test/in_app_purchase_test.dart:4]: Haha thanks! I was probably dreaming then. Will fix.

File: packages/in_app_purchase/lib/src/billing_client_wrappers/purchase_wrapper.dart [hunk 19]
```diff
@@ -168,16 +185,20 @@ class PurchaseHistoryRecordWrapper {
   /// device"](https://developer.android.com/google/play/billing/billing_library_overview#Verify-purchase-device).
   /// Note though that verifying a purchase locally is inherently insecure (see
   /// the article for more details).
+  @JsonKey(defaultValue: '')
   final String originalJson;
 
   /// The payload specified by the developer when the purchase was acknowledged or consumed.
-  final String developerPayload;
+  ///
+  /// If the value is `null` if it wasn't specified when the purchase was acknowledged or consumed.
```
Comment (stuartmorgan) [inline packages/in_app_purchase/lib/src/billing_client_wrappers/purchase_wrapper.dart:193]: Nit: s/If the value/The value/

File: packages/in_app_purchase/lib/src/store_kit_wrappers/sk_payment_queue_wrapper.dart [hunk 20]
```diff
@@ -124,49 +125,51 @@ class SKPaymentQueueWrapper {
...
+    final SKTransactionObserverWrapper observer = _observer!;
     switch (call.method) {
       case 'updatedTransactions':
         {
           final List<SKPaymentTransactionWrapper> transactions =
               _getTransactionList(call.arguments);
           return Future<void>(() {
-            _observer.updatedTransactions(transactions: transactions);
+            return observer.updatedTransactions(transactions: transactions);
```
Comment (stuartmorgan) [inline packages/in_app_purchase/lib/src/store_kit_wrappers/sk_payment_queue_wrapper.dart:151]: Why are you adding a return here, but not in any of the other cases?

File: packages/in_app_purchase/pubspec.yaml [hunk 21]
```diff
@@ -36,5 +32,5 @@ flutter:
         pluginClass: InAppPurchasePlugin
 
 environment:
-  sdk: ">=2.3.0 <3.0.0"
+  sdk: ">=2.12.0 <3.0.0"
```
Comment (stuartmorgan) [inline packages/in_app_purchase/pubspec.yaml:35]: Based on the latest guidance from https://dart.dev/null-safety/migration-guide#sdk-constraints I believe this should be `2.12.0-259.9.beta`. (Technically it says you should use whatever you tested with, but I think for maximal compatibility using the oldest nullability-stable version is the right thing for this repo right now.)

**Review by stuartmorgan — APPROVED** [review]
Minor nits, but otherwise LGTM!

File: packages/in_app_purchase/example/lib/main.dart [hunk 22]
```diff
@@ -254,8 +254,7 @@ class _MyAppState extends State<_MyApp> {
                     onPressed: () {
                       PurchaseParam purchaseParam = PurchaseParam(
                           productDetails: productDetails,
-                          applicationUserName: null,
-                          sandboxTesting: true);
```
[the remark this answers is not in the document]
Comment (cyanglaz) [reply packages/in_app_purchase/example/lib/main.dart:258]: The general testing flow actually doesn't need to include `simulateAskToBuy`. "Ask to buy" is only used to test the parental control popup, which prevents the general payment flow to be tested.
@LHLL is working on a patch to add a button to test "ask to buy" flow specifically and we will land it post NNBD

File: packages/in_app_purchase/lib/src/store_kit_wrappers/sk_payment_queue_wrapper.dart (hunk 20 above)
Comment (cyanglaz) [reply packages/in_app_purchase/lib/src/store_kit_wrappers/sk_payment_queue_wrapper.dart:151]: oh it's probably something i added when i tried to test a different return type and forgot to remove. Will remove

File: packages/in_app_purchase/pubspec.yaml (hunk 21 above)
Comment (cyanglaz) [reply packages/in_app_purchase/pubspec.yaml:35]: SG. Do we also update the example app to have the same version constraint? (>= 2.12.0-259.9.beta)

Comment (stuartmorgan) [reply packages/in_app_purchase/pubspec.yaml]: We may as well. I don't see it mattering if we forget for some examples, but it wouldn't hurt to be consistent.

File: packages/in_app_purchase/lib/src/store_kit_wrappers/sk_payment_transaction_wrappers.dart [hunk 23]
```diff
@@ -188,8 +196,8 @@ class SKPaymentTransactionWrapper {
   String toString() => _$SKPaymentTransactionWrapperToJson(this).toString();
 
   /// The payload that is used to finish this transaction.
-  Map<String, String> toFinishMap() => {
+  Map<String, String?> toFinishMap() => <String, String?>{
         "transactionIdentifier": this.transactionIdentifier,
```
Comment (cyanglaz) [inline packages/in_app_purchase/lib/src/store_kit_wrappers/sk_payment_transaction_wrappers.dart:200]: @stuartmorgan I updated here so when `transactionIdentifier` is null, the map contains null. Because transactionIdentifier is nullable, the platform code is expecting a null value from the map.
I also added tests for this.

**Review by cyanglaz — COMMENTED** [review]
@stuartmorgan Updated per last comments, PTAL

**Review by stuartmorgan — APPROVED** [review]
LGTM

**Comment by cyanglaz** [comment]
@stuartmorgan Got a publishable warning because the package is stable version and depends on a pre-release version of dark sdk 
```
Package validation found the following potential issue:
* Packages with an SDK constraint on a pre-release of the Dart SDK should themselves be published as a pre-release version. If this package needs Dart version 2.12.0-259.9.beta, consider publishing the package as a pre-release instead.
  See https://dart.dev/tools/pub/publishing#publishing-prereleases For more information on pre-releases.
Package validation found the following hint:
* You're about to publish a package that opts into null safety.
  The previous version (0.3.5+1) isn't opted in.
  See https://dart.dev/null-safety/migration-guide for best practices.

Package has 1 warning and 1 hint..
pub finished with exit code 65
Unable to publish in_app_purchase
�[31mFAIL: The following 1 package(s) failed the publishing check:
LocalDirectory: '/tmp/cirrus-ci-build/packages/in_app_purchase'�[0m
```
Should we revert the environment dependency to stable?

**Comment by stuartmorgan** [comment]
Looks like you just need to remove the `.` from https://github.com/flutter/plugins/blob/master/script/tool/lib/src/publish_check_command.dart#L100

Feel free to do that in this PR.

**Comment by cyanglaz** [comment]
Landing on red, attempt to fix the tree.

**Comment by stuartmorgan** [comment]
I was publishing my last round of NNBD changes this morning, so published this as well.
